content/en/continuous_integration/troubleshooting.md

```diff
@@ -3,20 +3,30 @@ title: Troubleshooting CI Visibility
 
+### Your tests are instrumented, but Datadog isn't showing any data
 
-### I instrumented my tests but I don't see any data in Datadog
+1. Go to the [Setup Tracing on CI Tests][9] page for the language you're instrumenting and check the _Compatibility_ section. Make sure the testing framework you are using is supported.
+2. Check if you see any test results in the [Test Runs][4] section. If you do see results there, but not in the [Tests][5] section, Git information is missing. See [Data appears in Test Runs but not Tests](#data-appears-in-test-runs-but-not-tests) to troubleshoot it.
+3. For languages other than Swift, make sure the Datadog Agent is running on the host where tests are run (accessible at `localhost:8126`), or if accessible on another hostname or port, make sure you run your tests with the appropriate Agent hostname set in the `DD_AGENT_HOST` and the appropriate port in `DD_TRACE_AGENT_PORT` environment variables. You can activate [debug mode][8] in the tracer to check if it's able to connect to the Agent.
+4. If you still don't see any results, [contact Support][3] for troubleshooting help.
 
-1. Check if you see any results in the [Test Runs][4] section. In the [Tests][5] section, only tests that contain git information will be shown.
-2. If you are executing your tests in a container, make sure you are forwarding the required CI provider environment variables to them. Check out the [Running tests inside a container][6] section for more information.
-3. For all languages except Swift, make sure the Datadog Agent is running in the host where tests are run (accessible at `localhost:8126`), or if accessible on another hostname and/or port, make sure you run your tests with the appropriate hostname set in the `DD_AGENT_HOST` and the appropriate port in `DD_AGENT_PORT` environment variables.
-4. If you still don't see any results, [contact Support][3] and we'll help you troubleshoot.
+### Data appears in Test Runs but not Tests
+
+If you can see test results data in the **Test Runs** tab, but not the **Tests** tab, Git metadata (repository, commit and/or branch) is probably missing. To confirm this is the case, open a test execution in the [Test Runs][4] section, and check that there is no `git.repository_url`, `git.commit.sha`, or `git.branch`. If these tags are not populated, nothing will show on the [Tests][5] section.
+
+1. Tracers will first try to fetch Git metadata using the local `.git` folder by executing `git` commands. This is the preferred approach, as it will populate all Git metadata fields, including commit message, author and committer information.
+2. If the `.git` folder is not present, or the `git` binary is not installed, tracers will fallback to use the environment variables set by the CI provider (as listed in the [Running tests inside a container][6] section) to collect Git information. This will populate, at least, repository, commit hash and branch information.
```

Comment:
```suggestion
2. If the `.git` folder is not present, or the `git` binary is not installed, tracers will fallback to use the environment variables set by the CI provider to collect Git information. See the [Running tests inside a container][6] page for a list of environment variables to set with the provider. This will populate, at least, repository, commit hash and branch information.
```

Reply:
The env vars are not to be set by the user, just read by the tracer. What this (or similar):
```suggestion
2. If the `.git` folder is not present, or the `git` binary is not installed, tracers will fallback to use the environment variables set by the CI provider to collect Git information. See the [Running tests inside a container][6] page for a list of environment variables that the tracer will attempt to read for each supported CI provider. This will populate, at least, repository, commit hash and branch information.
```